src/zmq.cpp

```diff
@@ -481,6 +481,19 @@ s_recvmsg (zmq::socket_base_t *s_, zmq_msg_t *msg_, int flags_)
     return (int) (sz < INT_MAX? sz: INT_MAX);
 }
 
+static int
+s_recvmsg_peer (zmq::socket_base_t *s_, void *zid, int zid_len, zmq_msg_t *msg_,
+                                                                int flags_)
+{
+    int rc = s_->recv_peer (zid, zid_len, (zmq::msg_t *) msg_, flags_);
+    if (unlikely (rc < 0))
+        return -1;
+
+    //  Truncate returned size to INT_MAX to avoid overflow to negative values
```

Comment:
this is a new API, so there's no need to constrain the return type. If it needs to be bigger than an int, make it so

Reply:
OK thanks. I was confused about return values.